specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/common.json

```diff
@@ -0,0 +1,362 @@
+      "enum": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_RAGRS",
+        "Standard_ZRS",
+        "Premium_LRS",
+        "Premium_ZRS",
+        "Standard_GZRS",
+        "Standard_RAGZRS"
+      ],
+      "x-ms-enum": {
+        "name": "SkuName",
+        "modelAsString": true
+      }
+    },
+    "Tier": {
+      "readOnly": true,
+      "type": "string",
+      "description": "The SKU tier. This is based on the SKU name.",
+      "enum": [
+        "Standard",
+        "Premium"
+      ],
+      "x-ms-enum": {
+        "name": "SkuTier",
+        "modelAsString": false
+      }
+    },
+    "CloudError": {
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody"
+        }
+      },
+      "description": "An error response from the Storage service."
+    },
+    "CloudErrorBody": {
+      "x-ms-external": true,
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for display in a user interface."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error. For example, the name of the property in error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "A list of additional details about the error."
+        }
+      },
+      "description": "An error response from the Storage service."
+    },
+    "SignedIdentifier": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "An unique identifier of the stored access policy."
+        },
+        "accessPolicy": {
+          "$ref": "#/definitions/AccessPolicy",
+          "description": "Access policy"
+        }
+      }
+    },
+    "AccessPolicy": {
+      "properties": {
+        "startTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Start time of the access policy"
+        },
+        "expiryTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Expiry time of the access policy"
+        },
+        "permission": {
+          "type": "string",
+          "description": "List of abbreviated permissions."
+        }
+      }
+    },
+    "StorageTaskReportSummary": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageTaskReportInstance"
+          },
+          "x-ms-identifiers": [],
```

Comment:
>  "x-ms-identifiers": [], [](http://example.com/codeflow?start=8&length=25)

they don't have a task id or anything?

Reply:
Rule

The default identifying property is id. You don’t need to add this annotataion if there is the id property in the object model.
If there is no appropriate identifying properties, leave it as an empty array.
You can provide multiple properties to form the composite identifier.
Use json pointer if the identifying property is an inner property. E.g., "x-ms-identifiers": ["/properties/subnet/id"]

Reply:
So the Id generated is system generated, it is read only resource and the name is system generated, customer cannot really do a GET hence the identifier is [] 
If you insist we can remove it, however , IMO it does not have a deterministic identifier and should not be included here

Reply:
Do you mean that 'being system generated' is a reason customer can't do a get? I don't think it necessarily is.

What do you mean by deterministic? Is it unstable over time? Or are these resources temporary?